generator/generator.yml

```diff
@@ -400,3 +400,44 @@ modules:
         type: EnumAsStateSet
     auth:
       community: raritan_public
+
+# InfraPower PDU's
+#
+# https://www.austin-hughes.com/support/software/infrapower/IPD-MIB.7z
+# The circuit table in the infrapower MIB is invalid as it has a table (circuitTable) nested inside another table (pduTable), and this invalid setup then confuses the generator.
+  infrapower_pdu:
+    walk:
+      - sysUpTime
+      - 1.3.6.1.4.1.34550.20.2.1.1.1.1 # pduIndex
+      - 1.3.6.1.4.1.34550.20.2.1.1.1.6 # pduName
+      - 1.3.6.1.4.1.34550.20.2.1.1.1.7 # pduLocation
+      - 1.3.6.1.4.1.34550.20.2.1.1.1.13 # pduMainLoadVoltage
+      - 1.3.6.1.4.1.34550.20.2.1.1.1.14 # pduMainLoadAmp
+      - 1.3.6.1.4.1.34550.20.2.1.1.1.17 # pduMainActivePower
+    lookups:
+      - source_indexes: [pduIndex]
+        lookup: pduName
+      - source_indexes: [pduIndex]
+        lookup: pduLocation
+    overrides:
+      pduIndex:
+        ignore: true # Lookup metric
+      pduName:
+        ignore: true # Lookup metric
+      pduLocation:
+        ignore: true # Lookup metric
+      pduMainLoadAmp:
+        regex_extracts:
+          '':
+            - regex: '(.*)(.)'
+              value: '$1.$2'
+      pduMainLoadVoltage:
+        regex_extracts:
+          '':
+            - regex: '(.*)(.)'
+              value: '$1.$2'
+      pduMainActivePower:
```

Comment:
You still need to convert this to watts

Reply:
Still unsure on how to do it, it needs to be /100000 yet its possible for their to not be that many digits in the output.

0= 00.00000kW
25 = 00.00025kW
123 = 00.00123kW
1234 = 00.01234kW
28819 = 00.28819kW
123456 = 01.23456Kw
1234567 = 12.34567kW

Reply:
The value is already in deci-watts, so we can apply the same regex as elsewhere.